### PR TITLE
adding per service slo config

### PR DESCRIPTION
**Pull request by yuriatgoogle**
Fixes #

## Proposed Changes

* added services module
* added ability to set per service latency goal and thresholds
* added latency thresholds for top 4 services based on historical data

**Release Note**

```release-note
Modified service and SLOs configurations for better configurability
```

### CODE REVIEW

File: terraform/alerting/module.service/main.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+resource "google_monitoring_custom_service" "service" {
+  service_id   = var.service_name
+  display_name = var.service_name
+  project      = var.project
+}
```
Comment (sethvargo) [inline terraform/alerting/module.service/main.tf:5]: Can you configure your editor to add a newline at EOF? Terraform will accept it, but it'll cause lots of unnecessary future diffs on PRs.

**Review by sethvargo — APPROVED** [review]
/lgtm

File: terraform/alerting/slos.tf [hunk 2]
```diff
@@ -14,59 +14,75 @@
...
+    latency_goal = 0.9
+    latency_threshold = 60000
   }
-  default_slo_thresholds = {
-    adminapi     = merge(local.default_per_service_slo, { enable_alert = true })
-    apiserver    = merge(local.default_per_service_slo, { enable_alert = true })
+  service_configs = {
+    adminapi     = merge(local.default_per_service_slo,
+                        { enable_alert = true ,
+                          latency_goal = 0.99 ,
```
Comment (yegle) [inline terraform/alerting/slos.tf:25]: Make this latency_goal part of the default?

File: terraform/alerting/slos.tf [hunk 3]
```diff
@@ -14,59 +14,75 @@
...
+                          latency_threshold = 6000 })
+    apiserver    = merge(local.default_per_service_slo,
+                        { enable_alert = true ,
+                          latency_goal = 0.99 ,
+                          latency_threshold = 400 })
```
Comment (yegle) [inline terraform/alerting/slos.tf:30]: The unit for this is ms. Is this service really hitting <400ms for 99% of the requests?

File: terraform/alerting/module.service/main.tf (hunk 1 above)
Comment (yuriatgoogle) [reply terraform/alerting/module.service/main.tf:5]: Updated VS Code setting for this, thank you.

File: terraform/alerting/slos.tf [hunk 4]
```diff
@@ -14,59 +14,75 @@
...
   notification_channels = google_monitoring_notification_channel.channels
 
-  for_each = merge(local.default_slo_thresholds, var.slo_thresholds_overrides)
+  for_each = merge(local.service_configs, var.slo_thresholds_overrides)
 
-  service_name = each.key
-  goal         = each.value.latency.goal
-  threshold    = each.value.latency.threshold
-  enable_alert = each.value.enable_alert
+  custom_service_id   = each.key
```
Comment (yegle) [inline terraform/alerting/slos.tf:83]: Hmm I doubt this works... It's a string like this: `projects/412869601512/services/verification-server`

File: terraform/alerting/slos.tf (hunk 2 above)
Comment (yuriatgoogle) [reply terraform/alerting/slos.tf:25]: Done, thank you.

File: terraform/alerting/slos.tf (hunk 3 above)
Comment (yuriatgoogle) [reply terraform/alerting/slos.tf:30]: Yep!
image

File: terraform/alerting/module.service/main.tf (hunk 1 above)
Comment (sethvargo) [reply terraform/alerting/module.service/main.tf:5]: You can also ask vscode to fmt for you too 👍

File: terraform/alerting/slos.tf (hunk 4 above)
Comment (yuriatgoogle) [reply terraform/alerting/slos.tf:83]: In my test, it worked - created the 8 services with 2 SLOs for each one:

image

File: terraform/alerting/slos.tf [hunk 5]
```diff
@@ -14,59 +14,71 @@
...
   default_per_service_slo = {
+    enable_alert      = false
     availability_goal = 0.995
-    latency = {
-      goal      = 0.9
-      threshold = 60000
-    }
-    enable_alert = false
+    latency_goal      = 0.99
+    latency_threshold = 60000
```
Comment (mariliamelo) [inline terraform/alerting/slos.tf:20]: Can you add in a comment the unit of this latency?

Comment (yuriatgoogle) [reply terraform/alerting/slos.tf]: Done.

**Review by sethvargo — APPROVED** [review]

**Pull request merged**